### DISCUSSION TITLE
Best way to do a multi-level menu

### DISCUSSION THREAD

**Discussion by holroy** [chhoumann/quickadd#493] [Q&A]
My use case is that I want to trigger a single command/choice on mobile, where I can choose between various options to insert certain predefined elements into my daily journal. These elements can either food (and kcal) oriented, activity oriented or task related. I'm not sure if possible, but I would also like the opportunity to add daily notes entries _not for current day_, and possibly to add extra text to a given element.

In most cases, and that is the main focus of this question is that I'm wondering how to build and maintain a larger structure which in most cases should render as a custom checklist item line.

Example of the menu I'm imagining:

- Food 
  - Breakfast
    - Alternative 1, 200 kcal
    - Alternative 2, 250 kcal
  - Lunch
    - Lunch alt 1, 300 kcal
    - lunch alt 2, 275 kcal
  - Dinner
    - ...
    - ...
  - Evening meal/snack
    - ...
- Activity
  - Discgolf
    - Course 1
    - Course 2
    - Course 3
    - ... 
  - Walking
    - Standard trip 1
    - Standard trip 2
    - ... 
    - **Custom entry? or just the basic**
  - Training/workout
- Evaluation
  - Score 1
  - Score 2
  - Score 3 
  - ...
  - Score 9        

For the _Food_ and _Activity_ elements, the typical output would be a complete line looking like the following:

```
- [F] Alternative 1, [kcal:: 400]
```

For stuff like the _Evaluation_, which might need to be in a different setting/command/choice, I would typically just like to add some text to the current line, like `[evaluation:: 3]`

----

So the question is how do I build such a structure in a way which is easy to maintain and keep updated. I'm not afraid of coding, so I could easily use the suggester, or a combination of elements where some of them are javascript segments. But I would like for some general pointers on how to build such a structure without having to create a new element within the QuickAdd settings for each and every menu element. (And even though I most likely could, I would like to avoid changing the `data.json` programmatically...

**Comment by holroy** [comment]
What I'm currently trying to do is 
- Multi: "My multi"
  - Multi: "Acitivity"
    -  Capture: "Discgolf" – Where the capture format is `{{MACRO: ...}`, to Macro "Choose discgolf course", which again is configured to execute the "chooseDiscgolfCourse.js", which in turn is a javascript doing a dataview query listing the options.

Some caveats with this approach:
- I've got to both define both the macro with a name, and then link it to the corresponding javascript file. Could this be done using inline scripts somehow?
- I'm loosing the "<- back" option when displaying the suggester query from my javascript. Is it possible to have that as part of a suggester prompt? How would I link it back into the other stuff
- If I ever want to change the _Capture_ options, and I've extended this syntax to all of my different groups (activitity, food, tasks stuff, evaluation stuff, ... ), I need to change each and every one of them. Especially problematic would that be for the food category, I think...

**Reply by chhoumann** [reply]
> I've got to both define both the macro with a name, and then link it to the corresponding javascript file. Could this be done using inline scripts somehow?

Yes! You should be able to basically copy-paste your current script into the inline script. Except, the first argument is passed in via `this`, so you need to use e.g. `this.quickAddApi`.

> I'm loosing the "<- back" option when displaying the suggester query from my javascript. Is it possible to have that as part of a suggester prompt? How would I link it back into the other stuff

This is a bit harder.
Hacky solution would be to run the `Discgolf` capture  / `Activity` multi again when you select `<- back`. You'd have to do that manually.

You can check this for inspiration: https://quickadd.obsidian.guide/docs/Examples/Macro_TogglManager
